client/src/main/java/io/cloudsoft/winrm4j/client/WinRmClient.java

```diff
@@ -124,8 +170,17 @@ private static String toDuration(long operationTimeout) {
         }
     }
 
-    private WinRmClient(URL endpoint) {
+    private WinRmClient(URL endpoint, String authenticationScheme) {
+        this.authenticationScheme = authenticationScheme != null ? authenticationScheme : AuthSchemes.BASIC;
         this.endpoint = endpoint;
+
+        // Needed to be async client according to http://cxf.apache.org/docs/asynchronous-client-http-transport.html
+        // TODO consider using async client for Basic authentication
+        if (this.authenticationScheme.equals(AuthSchemes.NTLM)) {
```

Comment:
I'd split all auth checks into `BASIC` and all the rest (which includes NTLM and Kerberos right now). The reason is that Kerberos is handled identically to NTLM, so there's no need to differentiate between the two.